### PR TITLE
Fix broken links, causing HTML proofer in circleci gates to fail

### PR DESCRIPTION
**Pull request by sdake**

### CODE REVIEW

**Comment by cmluciano** [comment]
This looks ok to me 

Only comment would be if it changed between releases such that it would break 0.5.X .

**Comment by ldemailly** [comment]
still having errors ?

**Comment by sdake** [comment]
@ldemailly Yes, with master of this repo (915e16e8b00fbe08693a12f63efd04b148f4faa5)
```
Checking 322 external links...
Ran on 233 files!


- ./_rakesite/test/help/troubleshooting.html
  *  External link https://raw.githubusercontent.com/istio/istio/master/install/kubernetes/webhook-create-signed-cert.sh failed: 404 No error
  *  External link https://raw.githubusercontent.com/istio/istio/master/install/kubernetes/webhook-patch-ca-bundle.sh failed: 404 No error
rake aborted!
HTML-Proofer found 2 failures!
```

**Comment by sdake** [comment]
@cmluciano I can't answer that question.  As the previous links were based upon master, I think this only improves the situation since those two links contents may have changed over time.

Cheers
-steve

**Comment by sdake** [comment]
@ldemailly nm, I misunderstood your question.  I think you meant that circleci is still failing with the same problem.  Perhaps I pushed the wrong branch.  Let me check and I'll address.

**Comment by sdake** [comment]
@ldemailly I noticed there is a directory `sdake@falkor-08:~/go/src/istio.io/istio.github.io/tmp$ ` which appears to not be removed between runs on a local system.  If circleci does not remove these cached entries, I think it is possible (pure speculation) that html-proofer could have some problem with the "removed links" from the cache.

After -rm -rf of the above tmp directory, I receive this error instead:

```
Found 0 links in the cache...
Adding 336 links to the cache...
Removing 0 links from the cache...
Checking 336 external links...
Ran on 233 files!


- ./_rakesite/test/docs/setup/kubernetes/sidecar-injection.html
  *  External link https://raw.githubusercontent.com/istio/istio/master/install/kubernetes/webhook-create-signed-cert.sh failed: 404 No error
  *  External link https://raw.githubusercontent.com/istio/istio/master/install/kubernetes/webhook-patch-ca-bundle.sh failed: 404 No error
- ./_rakesite/test/docs/tasks/telemetry/using-istio-dashboard.html
  *  External link https://hub.docker.com/r/grafana/grafana failed: got a time out (response code 0)
rake aborted!
HTML-Proofer found 3 failures!
/usr/local/bundle/gems/html-proofer-3.7.3/lib/html-proofer/runner.rb:167:in `print_failed_tests'
/usr/local/bundle/gems/html-proofer-3.7.3/lib/html-proofer/runner.rb:48:in `run'
/srv/jekyll/Rakefile:22:in `block in <top (required)>'
/usr/local/bundle/gems/rake-12.3.0/exe/rake:27:in `<top (required)>'
Tasks: TOP => test
(See full trace by running task with --trace)
```

which seems to indicate the bindmount cache contents are in some way affecting the results (2 errors prior, 3 errors now).

**Comment by sdake** [comment]
@ldemailly I am not familiar with the Ruby language, and could use a bit of help getting this PR functional.

Cheers
-steve

**Comment by sdake** [comment]
@ldemailly I guess I was confused by the expected behavior of HTML proofer and didn't run it multiple times.  This PR now works locally.  During the process of understanding how html proofer works - removed tmp and _rakesite (workflow was docker rm jekyll - docker run jekyll.

I would have expected the tool to produce a list of all failed usages of references.  Instead it displays teh first reference it finds.  That is fine - just not what I expected.

Cheers
-steve

**Comment by sdake** [comment]
@ldemailly this probably needs to be cherry-picked to the 0.7 branch - or folks that use the automatic sidecar injection documentation will find dead links in the 5 places the links are used.  Github doesn't offer redirects.  Which workgroup is responsible for that?  Test and release?

Cheers
-steve

**Comment by ldemailly** [comment]
Have we made 0.7 branch yet?

File: _docs/setup/kubernetes/sidecar-injection.md [hunk 1]
```diff
@@ -148,7 +148,7 @@ secret for the sidecar injector webhook to consume.
 
 _Note_: Kubernetes CA approval requires permissions to create and approve CSR. See 
 [https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster ](https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/) and 
-[install/kubernetes/webhook-create-signed-cert.sh](https://raw.githubusercontent.com/istio/istio/master/install/kubernetes/webhook-create-signed-cert.sh) for more information.
+[install/kubernetes/webhook-create-signed-cert.sh](https://raw.githubusercontent.com/istio/istio/release-0.7/install/kubernetes/webhook-create-signed-cert.sh) for more information.
```
Comment (ldemailly) [inline _docs/setup/kubernetes/sidecar-injection.md]: This file should be in the release tgz

**Comment by sdake** [comment]
@ldemailly sweet - I didn't actually check the 0.7 branch was created, and it hasn't been branched yet, so no need for backport.

Cheers
-steve

File: _docs/setup/kubernetes/sidecar-injection.md (hunk 1 above)
Comment (sdake) [reply _docs/setup/kubernetes/sidecar-injection.md]: curl on that link works - and it matches the existing intent of the document.  What I didn't want to do was get into a document rewrite to unblock the gate. :)

Not sure if github does automatic tar/gz removal - I'll give it a try and push a new PR if you think it is necessary.

Cheers
-steve

**Pull request merged**